### PR TITLE
Authentication Flow V1.0

### PR DESCRIPTION
**Pull request by mahid797**
# **Authentication Flow V1.0 - Issue #2**

This version will require significant refactoring for performance improvements, backend integration, and functionality once the backend is up and running, hence V1. However, the UI is fully built and aligns with the Figma designs.

### Pages implemented:

1. **Sign In Page**  

image
---
2. **Sign Up Page** 

image
---
3. **Forgot Password Page** 

image
---
4. **Password Reset Email Sent Page**  

image
---
5. **Set New Password Page**

image
---
6. **Password Reset Confirmation Page**

image
---
7. **Successful Account Creation Page**  

image
---

**Note:** This is just the initial version, focused on building out the UI. Future iterations will refine performance, error handling, and backend integration as we move forward.

### CODE REVIEW

**Comment by mahid797** [comment]
Fixed the `.env` issue as the file was still exposed in the master branch

File: Client/src/app/auth/sign-up/page.tsx [hunk 1]
```diff
@@ -1,4 +1,254 @@
...
+				<Box
+					component="form"
+					onSubmit={handleSubmit}
+					noValidate
+					minWidth={400}
+					gap={3}
+					display={'flex'}
+					flexDirection={'column'}>
+					{/* First Name */}
+					<FormLabel htmlFor="firstName">
```
Comment (Paxxack) [inline Client/src/app/auth/sign-up/page.tsx:86]: I noticed that the code from lines 86 to 108 could be refactored into a separate component. This change could potentially reduce the current 120 lines of code to about 10-20 lines, promoting a more 'DRY' (Don't Repeat Yourself) approach. It would enhance the overall maintainability of our codebase.

File: Client/src/app/auth/sign-up/page.tsx [hunk 2]
```diff
@@ -1,4 +1,254 @@
...
+	FormLabel,
+	TextField,
+	Typography,
+} from '@mui/material';
+import { useRouter } from 'next/navigation';
+import { ChangeEvent, FormEvent, useState } from 'react';
+import CheckIcon from '../../../../public/assets/icons/auth/CheckIcon';
+import Title from '../../../../public/assets/icons/sidebar/SidebarTitle';
+
+export default function SignUp() {
```
Comment (Paxxack) [inline Client/src/app/auth/sign-up/page.tsx:16]: We could enhance performance by consolidating most of those state declarations into a single object. This change would streamline our state management and improve overall efficiency. For example, we could use the following approach:

```
const [signUpData, setSignUpData] = useState({
        firstName: '',
        lastName: '',
        email: '',
        password: '',
        confirmPassword: '',
    });

File: Client/src/components/sidebar/DropdownMenu.tsx
Comment (Paxxack) [inline Client/src/components/sidebar/DropdownMenu.tsx:1]: I understand that it’s easy to get carried away when working on a project, but incorporating refactoring, bug fixes, or unrelated changes can make the PR less clear and larger than necessary. Keeping our PRs clean and concise will greatly benefit our review process. Thank you for your understanding

**Review by Paxxack — APPROVED** [review]
I do understand, as previously discussed, that the changes will be applied to your next version but I wanted to keep track of those issues for later review.
